### PR TITLE
Example for MemPooling.

### PR DESCRIPTION
**Pull request by wsad1**
#2623.
Added a working example for MemPooling. This example demonstrates 2 ideas the original paper uses to stabilize training.

1. The centroids `k` are updated based on the gradients of `kl_loss` at the end of every epoch. All other parameters are updated batch wise based on classification loss and epoch wise on `kl_loss`
2. The learning rate is decayed every 500 epochs.

### CODE REVIEW

**Comment by wsad1** [comment]
@rusty1s , let me know if you need some changes made. On the other hand if you feel this example is not necessary i can close the PR.

**Comment by rusty1s** [comment]
It's definitely needed and highly appreciated. Will probably have time to merge this on the weekend.

**Comment by wsad1** [comment]
Thanks @rusty1s for all your work. Please take your time with the PR.

**Comment by rusty1s** [comment]
Thank you!
